components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java

```diff
@@ -89,4 +91,33 @@ public UserSession getSession(String sessionId) throws SessionManagementServerEx
         }
         return null;
     }
+
+    private void getApplicationFromAppID(Application application) throws SessionManagementServerException {
+
+        Connection identityDBConnection = JDBCPersistenceManager.getInstance().
+                getDBConnection(false);
+        try {
```

Comment:
shall we use try with resource and close all the connections, prepared statements and result sets properly.

Reply:
Or we can use `JdbcTemplate` since it is already used in the other method.